File: PrintArr.java
/**
 * PrintArr
 */
public class PrintArr {
    public static void printSubsets(int input[],int index,String output) {
		// Write your code here
        if(index == input.length)
        {
            System.out.println(output);
            return;
        }
        printSubsets(input,index+1,output);
        printSubsets(input,index+1,output+" "+input[index]);
    }
    public static void main(String[] args) {
        int arr[] = {15,20,12};
        printSubsets(arr,0,"");
    }
}

File: print_arr.cpp
#include <iostream>
#include<string>
using namespace std;

void print_sub(int *arr,int index,string output)
{
	if(index == arr.size())
	{
		cout<< output<<endl;
		return;
	}
	print_sub(arr,index+1,output);
	print_sub(arr,index+1,output+to_string(arr[index]));
}

int main()
{
    int arr[] = {1,2,3};
	print_sub(arr,0,"a");
	return 0;
}
